File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count, element = 0,0
        for i in nums:
            if(count==0):
                element = i
            if(i==element):
                count += 1
            else:
                count -= 1
        return element

File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majority = nums[0], count =1;
        for(int i = 1; i<nums.size();i++){
            if(nums[i] != majority){
                count--;
                if(count == 0){
                    count = 1;
                    majority = nums[i];
                }
            } else {
                count++;
            }
        }
        return majority;
    }
};